### ISSUE TITLE
y0ng0p3 - The Midas protocol can consume stale price data or cannot operate on some EVM chains

### ISSUE THREAD

**Issue by sherlock-admin4** [sherlock-audit/2024-05-midas-judging#92]
y0ng0p3

medium

# The Midas protocol can consume stale price data or cannot operate on some EVM chains

## Summary

The stale period of  3 days is too large for Ethereum and Arbitrum chains, leading to consuming stale price data.

## Vulnerability Detail

In the `DataFeed` contract, the [`_HEALTHY_DIFF` constant is set to **3 days**](https://github.com/sherlock-audit/2024-05-midas/blob/main/midas-contracts/contracts/feeds/DataFeed.sol#L27). This means the `DataFeed::_getDataInBase18` function would consider the price data fed by Chainlink's price feed aggregators to be stale only after the last update time has elapsed **3 days**.

The [documentation](https://github.com/sherlock-audit/2024-05-midas/blob/main/README.md?plain=1#L52C1-L53C249) states that the protocol trusts Chainlink to __always__ return a good price if it has not been 3 days since the last update. But this is only true (clearly stated) for the `IB01/USD` feed which, according to the [documentation](https://github.com/sherlock-audit/2024-05-midas/blob/main/midas-contracts/README.md?plain=1#L85), is not yet implemented in contracts.
Since this is not specified, I legitimately assume that the Midas protocol does not 100% trust Chainlink's `EUR/USD` price to __never__ been stale if the time since the last update is no more than 3 days.

Since the Midas protocol supports Ethereum and Arbitrum chains, let's consider the `EUR/USD` Chainlink's feed on these chains:

- On Ethereum, Chainlink updates the price data [every ~24 hours](https://docs.chain.link/data-feeds/price-feeds/addresses?network=ethereum&page=1&search=eur). _HEALTHY_DIFF is **3 times** the aggregator heartbeat.
- On Arbitrum, Chainlink updates the price data [every ~1 hour](https://docs.chain.link/data-feeds/price-feeds/addresses?network=arbitrum&page=1&search=eur). _HEALTHY_DIFF is **72 times** the aggregator heartbeat.

On the chains the protocol is compatible with, **3 days** can be considered too large for the stale period, causing the `DataFeed::_getDataInBase18`` function very likely to return stale price data.

## Impact

Setting a stale period (_HEALTHY_DIFF) too large could lead to incorrect price of EURs. This can cause the `DepositVault::deposit` function to operate incorrectly, making the protocol disrupting.

## Code Snippet

https://github.com/sherlock-audit/2024-05-midas/blob/main/midas-contracts/contracts/feeds/DataFeed.sol#L27
https://github.com/sherlock-audit/2024-05-midas/blob/main/midas-contracts/contracts/feeds/DataFeed.sol#L75

## Tool used

Manual Review

## Recommendation

I recommend making `_HEALTHY_DIFF` just private, setting its value respectively in the `DataFeed::initialize` function and adding a **setter** function for updating its value.
When implementing the setter function, consider the role management in the protocol.

Duplicate of #110

**Comment by WangSecurity** [comment]
This report will be invalidated based on the escalation on #110.
